src/pages/blog/experimental-server-side-rendering.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: '../../layouts/Post.astro'
+title: 'Server-side Rendering with Astro'
+description: 'Announcing experimental support for server-side rendering in Astro'
+publishDate: 'April 5, 2022'
+socialImage: '/assets/blog/experimental-ssr/ssr.png'
+coverImage: '/assets/blog/experimental-ssr/ssr.png'
+lang: 'en'
+authors:
+  - matthew
+---
+
+# Server-side Rendering with Astro
+
+Today we are thrilled to announce experimental support for server-side rendering (SSR) in Astro. With SSR in Astro, we’ve unlocked entirely new use-cases (E-commerce, anyone?) and scale that just weren’t possible before. Our SSR implementation combines Astro’s unique server-first approach with dynamic features like user authentication, login flows, database access, data-fetching and more.
+
+If you’re ready to jump in, head over to [the docs](https://docs.astro.build/en/guides/server-side-rendering/) and get started.
+
+# Motivation
+
+We heard you! One of the questions we get most often is “Can Astro do SSR?”
+
+When we set out to build Astro, we decided to focus on the prebuilt, static site architecture popularly called “Jamstack” (also known static-site generation or SSG). This approach let us iterate and launch new features quickly, but came with some important limitations:
+
+- Large sites can’t statically pre-render to static HTML without introducing longer build times or slower, client-side rendering of dynamic content.
+- Dynamic sites can’t pre-render to static HTML without forcing you to rebuild and redeploy some portion of your site whenever your data changes.
+- Interactive sites can’t pre-render to static HTML without moving all user-individualized data to the client. User names, avatars, and permissions all get more complicated when you’re dealing with static, pre-built HTML.
+
+E-commerce is especially hurt by all three of the above problems. In an industry where every millisecond of load performance counts, how can you get the awesome performance of static HTML without the limitations?
+
+From Day 1, we knew that Astro’s approach to island architecture was uniquely positioned to solve this problem. Islands allow pages to load quickly and become interactive lazily; the user sees and is able to interact with the most important parts of the page, faster.
+
+# Enter: Server-Side Rendering (SSR)
+
+Server-side rendering is a well-understood solution to scaling large websites, going back to the earliest days of the internet. SSR is no silver bullet, but, when done right, SSR is an invaluable tool for certain use-cases.
+
+Take user authentication, which is something not easily achieved on a static site. You might do a couple of different ways:
+
+1. Check for a cookie or JWT in localStorage, then redirect to a login page if the user doesn’t have one. If using a cookie this prevents using `HttpOnly`, opening you to malicious JavaScript attacks. Also not shown here is, what do you do if the token is not valid? You still need to check and respond in that case.
```

Comment:
```suggestion
1. Check for a cookie or JWT in localStorage, then redirect to a login page if the user doesn’t have one. If using a cookie this prevents using `HttpOnly`, opening you to malicious JavaScript attacks.
```

Nit: this detail seemed to follow the code snippet. Maybe we can move this to the end?